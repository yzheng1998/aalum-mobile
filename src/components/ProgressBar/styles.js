import styled from 'styled-components'
import { themeGet } from 'styled-system'

export const EmptyBar = styled.View`
  background-color: ${themeGet('colors.grey')};
  height: 4px;
`
