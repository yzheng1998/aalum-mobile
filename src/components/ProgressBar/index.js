import React, { Component } from 'react'
import { EmptyBar } from './styles'
import LinearGradient from 'react-native-linear-gradient'
import theme from '../../../theme'

const { gradientLeft, gradientRight } = theme.colors

export default class ProgressBar extends Component {
  render() {
    const { progress } = this.props
    return (
      <EmptyBar>
        <LinearGradient
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
          colors={[gradientLeft, gradientRight]}
          style={{ width: progress, height: 4 }}
        />
      </EmptyBar>
    )
  }
}
