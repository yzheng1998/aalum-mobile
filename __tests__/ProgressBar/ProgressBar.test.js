import React from 'react'
import ProgressBar from '../../src/components/ProgressBar'
import mountWithTheme from '../helpers/theme'

describe('ProgressBar', () => {
  test('renders correctly', () => {
    const wrapper = mountWithTheme(<ProgressBar />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ progress: 0.89 })
    expect(wrapper).toMatchSnapshot()
  })
})
