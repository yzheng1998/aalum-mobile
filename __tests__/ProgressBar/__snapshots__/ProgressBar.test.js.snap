// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <ProgressBar>
    <Styled(View)>
      <StyledNativeComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(View)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  background-color: ",
                [Function],
                ";
  height: 4px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#A8A8A8",
                "height": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#A8A8A8",
                  "height": 4,
                },
              ]
            }
          >
            <LinearGradient
              colors={
                Array [
                  "#DA2752",
                  "#E935F8",
                ]
              }
              end={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              start={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "height": 4,
                  "width": undefined,
                }
              }
            >
              <_class
                colors={
                  Array [
                    4292486994,
                    4293473784,
                  ]
                }
                endPoint={
                  Object {
                    "x": 1,
                    "y": 0,
                  }
                }
                locations={null}
                startPoint={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                style={
                  Object {
                    "height": 4,
                    "width": undefined,
                  }
                }
              >
                <BVLinearGradient
                  colors={
                    Array [
                      4292486994,
                      4293473784,
                    ]
                  }
                  endPoint={
                    Object {
                      "x": 1,
                      "y": 0,
                    }
                  }
                  locations={null}
                  startPoint={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  style={
                    Object {
                      "height": 4,
                      "width": undefined,
                    }
                  }
                />
              </_class>
            </LinearGradient>
          </View>
        </View>
      </StyledNativeComponent>
    </Styled(View)>
  </ProgressBar>
</ThemeProvider>
`;

exports[`ProgressBar renders correctly 2`] = `
<ThemeProvider
  progress={0.89}
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <ProgressBar>
    <Styled(View)>
      <StyledNativeComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(View)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  background-color: ",
                [Function],
                ";
  height: 4px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#A8A8A8",
                "height": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#A8A8A8",
                  "height": 4,
                },
              ]
            }
          >
            <LinearGradient
              colors={
                Array [
                  "#DA2752",
                  "#E935F8",
                ]
              }
              end={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              start={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Object {
                  "height": 4,
                  "width": undefined,
                }
              }
            >
              <_class
                colors={
                  Array [
                    4292486994,
                    4293473784,
                  ]
                }
                endPoint={
                  Object {
                    "x": 1,
                    "y": 0,
                  }
                }
                locations={null}
                startPoint={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                style={
                  Object {
                    "height": 4,
                    "width": undefined,
                  }
                }
              >
                <BVLinearGradient
                  colors={
                    Array [
                      4292486994,
                      4293473784,
                    ]
                  }
                  endPoint={
                    Object {
                      "x": 1,
                      "y": 0,
                    }
                  }
                  locations={null}
                  startPoint={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  style={
                    Object {
                      "height": 4,
                      "width": undefined,
                    }
                  }
                />
              </_class>
            </LinearGradient>
          </View>
        </View>
      </StyledNativeComponent>
    </Styled(View)>
  </ProgressBar>
</ThemeProvider>
`;
